src/topology/algebra/ordered.lean

```diff
@@ -2178,22 +2179,140 @@ lemma tendsto_of_monotone {ι α : Type*} [preorder ι] [topological_space α]
 if H : bdd_above (range f) then or.inr ⟨_, tendsto_at_top_csupr h_mono H⟩
 else or.inl $ tendsto_at_top_at_top_of_monotone' h_mono H
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Ioi a)) (nhds_within (a⁻¹) (Iio (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within a (Iio a)) (nhds_within (a⁻¹) (Ioi (a⁻¹))) :=
 (continuous_inv.tendsto a).inf $ by simp [tendsto_principal_principal]
 
-@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Ioi_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Ioi (a⁻¹))) (nhds_within a (Iio a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Ioi _ _ _ _ (a⁻¹)
 
-@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv {α : Type*} [ordered_comm_group α]
+@[to_additive] lemma tendsto_inv_nhds_within_Iio_inv [ordered_comm_group α]
   [topological_space α] [topological_group α] {a : α} :
   tendsto has_inv.inv (nhds_within (a⁻¹) (Iio (a⁻¹))) (nhds_within a (Ioi a)) :=
 by simpa only [inv_inv] using @tendsto_inv_nhds_within_Iio _ _ _ _ (a⁻¹)
+
+lemma nhds_cleft_sup_nhds_cright (a : α) [topological_space α] [linear_order α] :
```

Comment:
Are those names `cleft` and `cright` documented somewhere?

Reply:
Not really, it's an idea of mine, but that's not very clear so I guess I'll take any name suggestion here

Reply:
Did you already used it elsewhere? I would rather add an extra letter to the punctured case, since `nhds` is not punctured. And it should be mentioned in the module docstring.

Reply:
No that's the only place I used them. So you would recommend using `left` for the closed case and `oleft` for the open case ? Would it be better to go back to interval based names for now, and return to it when we decided of a common naming scheme for all the left and right things ?

Reply:
I recommend `left` for the "closed" case but I don't really know what I recommend for the punctured case. Let's see what @urkud or @sgouezel think (we need a review from either of them anyway). If they don't care then let's not  worry too much about this name.

Reply:
What do you think about prime for punctured?

Reply:
Yes, why not?

Reply:
The problem I'd have with prime : how should I name the lemma I called `nhds_oleft_sup_nhds_cright` ? `nhds_left'_sup_nhds_right` is ugly, and very impractical (e.g double clicking in VSCode stops the selection at the prime symbol)

Reply:
The VS Code issue with primes in the middle of names might be fixable by editing the `editor.wordSeparators` setting for Lean files. Related: #3186.

Reply:
Oh then I guess it's okay, I'll use it for now